contracts/impl/EnigmaCommon.sol

```diff
@@ -87,6 +90,21 @@ library EnigmaCommon {
         return (_message.concat(uint64(_var.length).toBytesFromUint64())).concat(_var);
     }
 
+    /**
+    * Append the length of a variable and the variable to an existing bytes buffer
+    *
+    * @param _message Bytes buffer being appended to
+    * @param _var Bytes representation of value that needs to be concatenated to existing buffer
+    * @return New bytes buffer
+    */
+    function appendMessageKM(bytes memory _message, bytes memory _var)
+    internal
+    pure
+    returns (bytes memory)
+    {
+        return ((_message.concat(hex"00")).concat(uint64(_var.length).toBytesFromUint64())).concat(_var);
```

Comment:
does the length include the 00 byte as well?

Reply:
no, does not, but i believe the solution on core end also works same way as this now. should be resolved!

Reply:
@AvishaiW: this appears to be the matching code in `core`:
https://github.com/enigmampc/enigma-core/blob/77f48bc744c7d15a30fb42940c9a7d4987dcb351/enigma-principal/enclave/src/epoch_keeper_t/nested_encoding.rs#L20

As we can see the `0` byte is not taken into account in computing the length. Is this the intended behavior? It is not clear to me either way. I'm investigating, but I want confirmation from either @AvishaiW or @moriaab before merging this.